avm/res/insights/data-collection-rule/modules/nested_roleAssignments.bicep

```
@@ -0,0 +1,76 @@
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Name of the Data Collection Rule to assign the role(s) to.')
+param dataCollectionRuleName string
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId(
```

```suggestion
  'Role Based Access Control Administrator': subscriptionResourceId(
```
While we're at it ...

---

On a different note, I wonder if we should keep the roles and formatting in the parent. The only reason I say so is that users need to know they exist to actually set them, which gets harder if they're in a nested file. Also, assuming I ever get somebody to help me merge [this](https://github.com/Azure/bicep-registry-modules/pull/706) PR, the roles would need to be in the parent... I think

---

Addressed in: https://github.com/Azure/bicep-registry-modules/pull/3211/commits/dd9ddf789ccb41bf73fa82992e84dfeb59ed51bd